pinot-common/src/main/java/org/apache/pinot/common/utils/request/RequestUtils.java

```diff
@@ -192,7 +192,7 @@ public static Expression getLiteralExpression(Object object) {
       return RequestUtils.getLiteralExpression(((Number) object).longValue());
     }
     if (object instanceof Float || object instanceof Double) {
-      return RequestUtils.getLiteralExpression(((Number) object).doubleValue());
+      return RequestUtils.getLiteralExpression(Double.parseDouble(object.toString()));
```

Comment:
Separate `instanceof Float` and `instanceof Double`. We don't need to do this conversion for `Double`. Please also add some comment about why we need to do this conversion

Reply:
done